src/ecdsa/ellipticcurve.py

```diff
@@ -92,14 +92,16 @@ def __init__(self, p, a, b, h=None):
             self.__h = h
 
     def __eq__(self, other):
-        if isinstance(other, CurveFp):
-            """Return True if the curves are identical, False otherwise."""
-            return (
-                self.__p == other.__p
-                and self.__a == other.__a
-                and self.__b == other.__b
-            )
-        return NotImplemented
+        """Return True if the curves are identical, False otherwise."""
+        return (
+            isinstance(other, CurveFp)
+            and self.__p == other.__p
+            and self.__a == other.__a
+            and self.__b == other.__b
+        )
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
```

Comment:
that special handling of `NotImplemented` also means that `__ne__` should be implemented as `not self == other` (then we also don't have to check for class of other, as the `==` will do that and handle NotImplemented correctly)

Reply:
Good call! Thanks for catching that.